src/submit.rs

```diff
@@ -476,4 +477,44 @@ impl<'a> Submitter<'a> {
         )
         .map(drop)
     }
+
+    /// Performs a synchronous cancellation request, similiar to [AsyncCancel](crate::opcode::AsyncCancel),
+    /// except that it completes synchronously.
+    ///
+    /// [`AsyncCancelFlags`](types::AsyncCancelFlags) may be supplied to indicate the strategy for
+    /// matching operations to cancel.
+    ///
+    /// Available since 6.0.
+    pub fn register_sync_cancel(
+        &self,
+        user_data: u64,
+        fd: RawFd,
+        timeout: Option<Duration>,
+        flags: types::AsyncCancelFlags,
+    ) -> io::Result<()> {
+        // Default to infinite timeout if a timeout is not provided.
+        let mut timespec = sys::__kernel_timespec {
```

Comment:
Interesting that because this is a synchronous call, there is no issue of having to hold the Timespec data in a 'static location as in the Timeout opcode, or as there would be in the AsyncCancel opcode (if that opcode handled more of the underlying feature set).

As a result, I would have opted for std::time::Duration being the type here but quininer already weighed in that consistency of type, even though one is by pointer and the other by value, was to be preferred.